src/TaskViewSection/Toolbar/ToolbarButtons.jsx

```diff
@@ -1,159 +1,158 @@
-import Button from "../../Components/Button";
-import { MainContext, ViewSectionContext } from "../../Contexts";
+import React, { useContext } from 'react'
+import Button from '../../Components/Button'
+import { ViewSectionContext } from '../../Contexts/Contexts'
+import { useTaskContext } from '../../Contexts/TaskContext'
 
 export default function ToolbarButtons({ setAlert }) {
-  const { taskList, setTaskList } = useContext(MainContext);
-  const { selectedTabCategory } = useContext(ViewSectionContext);
+    const { taskList, setTaskList } = useTaskContext()
+    const { selectedTabCategory } = useContext(ViewSectionContext)
+    const allTabIsActive = selectedTabCategory === 'all'
+    const removeFromList = () =>
+        setTaskList(taskList.filter((item) => item.done !== true))
 
-  const isAnyTaskDone = () => {
-    const doneTasks = taskList.filter((item) => item.done === true);
-    for (const key of taskList) {
-      if (selectedTabCategory === "all" && doneTasks.length >= 1) {
-        setTaskList(taskList.filter((item) => item.done !== true));
-        return true;
-      }
+    const removeTasksHandler = () => {
+        for (const task of taskList) {
+            const isDone = task.done === true
+            const belongsToActiveTab = task.category === selectedTabCategory
 
-      if (key.category === selectedTabCategory && key.done === true) {
-        setTaskList(taskList.filter((item) => item.done !== true));
-        return true;
-      }
+            if (allTabIsActive || (belongsToActiveTab && isDone)) {
+                removeFromList()
+                showAlert({
+                    title: 'All done tasks has been removed successfully',
+                    type: 'success',
+                    background: 'bg-success',
+                    isShowed: true,
+                })
+            } else {
+                showAlert({
+                    title: 'There are no completed tasks to be deleted',
+                    type: 'error',
+                    background: 'bg-error',
+                    isShowed: true,
+                })
+            }
+        }
     }
-    return false;
-  };
 
-  const showAlert = (params) => {
-    setAlert({
-      title: params.title,
-      type: params.type,
-      background: params.background,
-      isShowed: params.isShowed,
-    });
-    setTimeout(() => {
-      setAlert({ isShowed: false });
-    }, 3000);
-  };
-  const removeTasksHandler = () => {
-    isAnyTaskDone()
-      ? showAlert({
-          title: "All done tasks has been removed successfully",
-          type: "success",
-          background: "bg-success",
-          isShowed: true,
+    const showAlert = (alertData) => {
+        setAlert({
+            title: alertData.title,
+            type: alertData.type,
+            background: alertData.background,
+            isShowed: alertData.isShowed,
+        })
+        setTimeout(() => {
+            setAlert({ isShowed: false })
+        }, 3000)
```

Comment:
question for you to check, what would happen if component will unmount before this timer will be fire